homeassistant/components/azure_event_hub/hub.py

```diff
@@ -0,0 +1,159 @@
+"""Hub for Azure Event Hub history."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from datetime import datetime
+import json
+import logging
+from typing import Any
+
+from azure.eventhub import EventData, EventDataBatch
+from azure.eventhub.aio import EventHubProducerClient
+from azure.eventhub.exceptions import EventHubError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL
+from homeassistant.core import Event, HomeAssistant, State
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.json import JSONEncoder
+from homeassistant.util.dt import utcnow
+
+from .client import AzureEventHubClient
+from .const import CONF_MAX_DELAY, CONF_SEND_INTERVAL, DEFAULT_MAX_DELAY, FILTER_STATES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AzureEventHub:
```

Comment:
It would be nice to move the hub in a separate PR without any other change to make it easier to see the important changes in this PR.

Reply:
ok, done!